docs/design/README.md

```diff
@@ -865,6 +865,9 @@ or restrictions on casts between pointers and integers.
 
 ### Arrays and slices
 
+**TODO:** The provisional array syntax documented here has been superseded by
+[p4682](/proposals/p4682.md).
```

Comment:
I see a couple link formats:

- `[p5168.md](/proposals/p5168.md)`
- `[p5168](/proposals/p5168.md)`

In "references" sections, I think we're using a different format:

-  `[#3762: Merging forward declarations](https://github.com/carbon-language/carbon-lang/pull/3762)`

IIRC, the reason for linking to the PR instead of the file was to be able to see the discussion, and the proposal name was considered helpful for context without needing to follow the link. Perhaps the same rationale would apply for these links?